Handler/Demo.hs

```diff
@@ -0,0 +1,37 @@
+module Handler.Demo where
+
+import Import
+
+import Control.Monad (filterM)
+import Data.Maybe (listToMaybe)
+
+getDemoR :: Handler Html
+getDemoR = do
+    root <- getAppRoot
+    siteId <- runDB $ findDemoSite root
+
+    defaultLayout $ do
+        setTitle "Carnival Demo"
+        $(widgetFile "demo")
+
+findDemoSite :: Text -> DB SiteId
+findDemoSite root = do
+    sites <- selectList [SiteBaseUrl ==. root] []
+    msite <- listToMaybe <$> filterM noMembers sites
+
+    maybe (createDemoSite root) (return . entityKey) msite
+
+  where
```

Comment:
Theoretically, then anyone could create a site with the root URL and it might become the site used by `/demo`. That just sounds dangerous in terms of abuse. I don't think there's anything a user could do now but who knows how `Site` evolves and what control of the demo site could mean in the future; enforcing that no one can administer this sites feels safer.

I'm probably being paranoid though. If you think it'd be fine, I can take it out. I assume you do?